### ISSUE TITLE
[Feature Request] Update Pint's tire diameter to be 868 instead of 837.86

### ISSUE THREAD

**Issue by koa28** [OnewheelCommunityEdition/OWCE_App#62]
Current Pint tire diameter as stated in [OWBaseBoard.cs](https://github.com/OnewheelCommunityEdition/OWCE_App/blob/21bdebadacd5951b2da3544cb1d16d57e80288ef/OWCE/OWCE/OWBaseBoard.cs#L135) is 837.86. However, when looking at the trip distance as reported in the OW App for a few of my rides, I get the following data points

Trip Odometer (revs) | Trip Miles 
---------------------|-----------
32502 | 17.5
24295 | 13.1
14287 | 7.7
18482 | 10.0

And if we plug in the original value of 837.86 using the formula `Revs * 837.86 * 0.001 * 0.001 * 0.6213` we end up with wrong miles estimated

It seems like OW uses 868 (or any value between 867 to 869) as the wheel diameter (which corresponds to an approximately 10.88 inch diameter tire, which ironically is the more like the diameter of a tire like the 11 inch XTraRad rather than the stock 10.5 Unili). Whatever their reasons, if we want to keep consistent with OW, we should probably use 868 as the diameter.

Table below shows the calculations with both values of Tire diameter

Trip Odometer (revs) | Trip Miles reported by App | Calculated based on 837.86 | Calculated based on 868
---------------------|----------------------------|------------------------------|-------------------------
32502 | 17.5 | 16.9 | 17.5 
24295 | 13.1 | 12.6 | 13.1
14287 | 7.7 | 7.4 | 7.7 
18482 | 10.0 | 9.6 | 10.0

**Comment by koa28** [comment]
Granted, this is a bit of a moot point currently, as the tire diameter isn't actually propagated to any of the converters, but that is a separate issue to resolve.

**Comment by beeradmoore** [comment]
Is it possible that official app is using the same tyre diameter for all boards?

**Comment by koa28** [comment]
I doubt it -- isn't the XR's tire 11.5 inches? In any case, would need someone with an XR to verify by looking at the app's trip page and then look at the number of revs in the diagnostics page.

**Comment by beeradmoore** [comment]
Yeah true, it would be way off for one board or another if it was wrong.

I never physically measured the tyre diameter or circumference. I had always just gone off 10.5inch, convert to millimetres, get the circumference of the circle. Maybe measured diameter is different? 

I am now curious:
- what would happen if you had a 10mi straight road what a google maps point to point would say.
- what your above tests yield for a Plus/XR (revs vs distance) (going to ask for some screenshots on [Facebook](https://www.facebook.com/groups/onewheelownersgroup/posts/2937698539852030/) for this one)

**Comment by koa28** [comment]
I probably won't be able to debug OW's settings for tire diameter, since I changed my tire to an XtraRad some time ago, so it's a different diameter tire anyway, and will not be much use to try and correlate OW's reported mileage (that is based on the stock tire) versus a GPS-reported one. I do in general find that a GPS-reported distance tends to be less than one calculated by number of revs, and someone pointed out before that the GPS won't take into account someone carving and traveling a "wavy" line on a path, as opposed to a perfectly straight line.

Having said that, is it more important for wheel calculations to match real-world, or match OW's estimates? If you match real-world, then you diverge from OW's distance calculations, eg their lifetime distance traveled is probably based on lifetime number of rotations multiplied by the number they use for tire diameter. Or if you want to estimate when pushback happens, it's probably at an RPM that is estimated to represent X mph rather than what is actually X mph.

**Comment by beeradmoore** [comment]
I did not get a single screenshot from my above FB request lol.

RE: GPS vs actual inaccuracy: there is also another one, does FM deduct milage that comes from wheel spins? I guess this is easy to test, milk crate it up and let it go and see if trip odometer ticks up in offical app.

But yeah on the "matching OWs estimates vs more accurate" matter. I think I'd prefer accurate over matching their figures. Assuming their figures are not real world accurate that is. If FM isn't using correct tyre measurements to calculate this then I am sure they have some reason for it and isn't an accident. But even getting correct trip odometer, things like #43 will still mean OWCE and FM apps can display different data because the board decided to misinform us.

Also curious now if you bring tyre pressure down to 10psi vs up high at 30psi is the odometer reading the same for both rides over the same path. Probably a less of a difference is 20psi tyre, 50lbs vs 300lbs riders. Maybe FM factor all this in and their value is a good estimate.

It was thought up, but there are currently no tasks carded up for it, but OWCE should allow users to enter their own tyre diameter. Especially useful if they change their tyre. I am curious as well if a "calculate my tyre diameter" feature is possible by recording GPS points (ideally in a straight line), measuring tyre rotations and then calculating correct value for this for that rider, with that tyre, at that PSI. Possibly the best technical solution, but also a lot of overkill.

**Comment by koa28** [comment]
Might be possible to calibrate where you need at least a certain amount of GPS accuracy before they can do their test run, and they need to travel in as straight line as possible, ie no carving. If the board sends its roll / yaw angles, you can ensure the board is at a certain level of level-ness. Maybe you can even do it intelligently during a ride where you detect that the conditions are right to start recording (good GPS signal, board level) and if you get a good number of rotations and distance traveled, you kinda "learn" the tire diameter as the ride goes along. Will be quite cool technically, though I'm not sure about the ROI of doing something like this.

**Comment by koa28** [comment]
I did some more experiments which lead to even more confusing numbers:

1. I physically measured the circumference of my XtraRad tire using a soft tape measure (the type a tailor would use) and it measured at around 835mm (which is actually really close to your calculated circumference of the stock pint tire despite the fact that the XtraRad is a 11 x 6 tire instead compared to the 10.5 x 6 stock pint tire). I tested this measurement at 15 PSI and 21 PSI, and there was no appreciable difference in measurement. Wonder what a similar measurement comes up to for the stock tire. 

2. I did a test run on a short route that I measured on google maps as 488.5 meters, and took the difference in trip revs at the start and end of the route, which came up to 620 revs, and if you do 488.5 / 620 * 1000 you get a calculated circumference of 788mm, which is much smaller than 835mm. The difference does appear very significant. The only thing I can think of is that I wasn't riding a perfectly straight line (I tried as far as possible to do it without carving). Maybe if the tire was always on an edge, it is using a smaller effective diameter? Or maybe the trip rev counter is a bit off? Or maybe Google Maps was off in the calculation? Might be better to use a running track with distance markers?

3. For the app that I deployed to my phone, I plugged in 870 for the wheel circumference, and it showed an identical trip mileage as the FM app -- both reporting that the trip took 1.4 miles. 

I mean, there are so many implications from this set of discrepancies. Eg when the FM app says you traveled 10 miles on 1 charge, how many miles did you really travel? Or when the Pint does a speed-limit pushback at 14.5mph (I presume the software uses a set rpm when it wants to limit your speed), was it really going 14.5mph? If we plug in 835 or 788 for the wheel diameter, is it going to show that the Pint's speed limit is more like 12-13mph?

**Comment by beeradmoore** [comment]
I measured my Plus at 895mm and my Pint at 865mm, both are stock.

865 does line up well with your 868 (to be fair I did use a braided USB cable and tape measure so I'll give it a few mm of accuracy in your favour).

The 895 is also very different from the calculated 917.66 that I'd expect the Plus to be. 

100 - (895 / 917.66 * 100) = 2.4693241506
100 - (865 / 837.86 * 100) = -3.2392046404

If I did this math right, we were 2.4% too small for Pint, and 3.2% too large for Plus. :| 

EDIT: So I did this math right and it lines up with what I see with [this](https://www.piday.org/calculators/circumference-calculator/#) online calculator.
circumference 895mm = diameter 11.21 inches
circumference 865mm = diameter 10.84 inches

I guess it was a bad idea assuming tyre size is an exact measurement and should have gone from real world measurements.

**Comment by koa28** [comment]
If you ride the Plus using the FM app, what trip revs do you get and what is the reported trip mileage? If you divide mileage by number of revs, it should give you the circumference that the app uses. See if it corresponds with the tire calculation that you measured.

I'm surprised that the XtraRad's circumference is lower than the stock Pint tire, since the tire is spec'ed as a 11 x 6 compared with the stock tire's 10.5 x 6.
